waspc/data/Generator/templates/sdk/wasp/auth/utils.ts

```diff
@@ -124,25 +124,39 @@ export async function updateAuthIdentityProviderData<PN extends ProviderName>(
   });
 }
 
-type FindAuthWithUserResult = {= authEntityUpper =} & {
+// PRIVATE API
+export type FindAuthWithUserResult = {= authEntityUpper =} & {
   {= userFieldOnAuthEntityName =}: {= userEntityUpper =}
 }
 
 // PRIVATE API
 export async function findAuthWithUserBy(
   where: Prisma.{= authEntityUpper =}WhereInput
-): Promise<FindAuthWithUserResult> {
-  return prisma.{= authEntityLower =}.findFirst({ where, include: { {= userFieldOnAuthEntityName =}: true }});
+): Promise<FindAuthWithUserResult | null> {
+  const result = await prisma.{= authEntityLower =}.findFirst({ where, include: { {= userFieldOnAuthEntityName =}: true }});
+
+  if (result === null) {
+    return null;
+  }
+
+  if (result.user === null) {
+    return null;
+  }
+
+  return result as FindAuthWithUserResult;
```

Comment:
I think preferrable options are:
```typescript
const result = await prisma.auth.findFirst({ where, include: { user: true } });
return result && result.user && { ...result, user: result.user };
```
Or:
```typescript
const result = await prisma.auth.findFirst({ where, include: { user: true } });

if (!result) {
  return null;
}

const { user, ...auth } = result;
return user && { user, ...auth };
```
I prefer the second one.

Reply:
I like the intention of not using type assertions. I went with:

```ts
const result = await prisma.auth.findFirst({ where, include: { user: true }});

if (result === null) {
  return null;
}

if (result.user === null) {
  return null;
}

return { ...result, user: result.user };
```